File: android/apolloui/src/main/java/io/muun/apollo/presentation/ui/fragments/landing/LandingFragment.java
package io.muun.apollo.presentation.ui.fragments.landing;

import io.muun.apollo.R;
import io.muun.apollo.presentation.ui.base.SingleFragment;
import io.muun.apollo.presentation.ui.view.MuunButton;

import android.os.Bundle;
import android.view.View;
import butterknife.BindView;
import butterknife.OnClick;
import com.airbnb.lottie.LottieAnimationView;

public class LandingFragment extends SingleFragment<LandingPresenter> {

    private static final String ARG_SHOW_ANIMATION = "show_animation";

    public static LandingFragment newInstance() {
        return new LandingFragment();
    }

    public static LandingFragment newInstanceWithAnimation() {
        final LandingFragment fragment = new LandingFragment();

        final Bundle arguments = new Bundle();
        arguments.putBoolean(ARG_SHOW_ANIMATION, true);
        fragment.setArguments(arguments);

        return fragment;
    }

    @BindView(R.id.signup_start)
    MuunButton startButton;

    @BindView(R.id.animation_view)
    LottieAnimationView lottieView;

    @Override
    protected void inject() {
        getComponent().inject(this);
    }

    @Override
    protected int getLayoutResource() {
        return R.layout.landing_activity;
    }

    @Override
    protected void initializeUi(View view) {
        super.initializeUi(view);

        if (getArgumentsBundle().getBoolean(ARG_SHOW_ANIMATION)) {
            lottieView.setAnimation(R.raw.logo_animation);
            lottieView.setMinFrame(40); // Small arbitrary correction to json exported animation

        } else {
            lottieView.setImageResource(R.drawable.wordmark_blue_beta_big_top);
        }
    }

    @OnClick(R.id.signup_start)
    public void onStartButtonClick() {
        startButton.setEnabled(false); // avoid double tap while preparing next Fragment
        presenter.startSignup();
    }

    @OnClick(R.id.login_start)
    public void onLoginButtonClick() {
        startButton.setEnabled(false); // avoid double tap while preparing next Fragment
        presenter.startLogin();
    }

    @Override
    public boolean onBackPressed() {
        finishActivity();
        return true;
    }
}

File: android/apolloui/src/main/java/io/muun/apollo/presentation/ui/fragments/landing/LandingFragment.kt
package io.muun.apollo.presentation.ui.fragments.landing

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.viewbinding.ViewBinding
import io.muun.apollo.R
import io.muun.apollo.databinding.LandingActivityBinding
import io.muun.apollo.presentation.ui.base.SingleFragment

class LandingFragment : SingleFragment<LandingPresenter>() {

    companion object {
        private const val ARG_SHOW_ANIMATION = "show_animation"

        @JvmStatic
        fun newInstance(): LandingFragment {
            return LandingFragment()
        }

        @JvmStatic
        fun newInstanceWithAnimation(): LandingFragment {
            val fragment = LandingFragment()
            fragment.arguments = Bundle().apply {
                putBoolean(ARG_SHOW_ANIMATION, true)
            }
            return fragment
        }
    }

    private val binding: LandingActivityBinding
        get() = getBinding() as LandingActivityBinding

    override fun bindingInflater(): (LayoutInflater, ViewGroup, Boolean) -> ViewBinding {
        return LandingActivityBinding::inflate
    }

    override fun inject() {
        component.inject(this)
    }

    override fun getLayoutResource(): Int =
        R.layout.landing_activity

    override fun initializeUi(view: View) {
        with(binding) {
            if (argumentsBundle.getBoolean(ARG_SHOW_ANIMATION)) {
                lottieView.setAnimation(R.raw.logo_animation)
                lottieView.setMinFrame(40) // Small arbitrary correction to json exported animation
            } else {
                lottieView.setImageResource(R.drawable.wordmark_blue_beta_big_top)
            }

            createWalletButton.setOnClickListener { onCreateWalletButtonClick() }
            recoverWalletButton.setOnClickListener { onRecoverWalletButtonClick() }
        }
    }

    private fun onCreateWalletButtonClick() {
        binding.createWalletButton.isEnabled = false // avoid double tap while preparing next Fragment
        presenter.startCreateWallet()
    }

    private fun onRecoverWalletButtonClick() {
        binding.recoverWalletButton.isEnabled = false // avoid double tap while preparing next Fragment
        presenter.startRecoverWallet()
    }

    override fun onBackPressed(): Boolean {
        finishActivity()
        return true
    }

    override fun setUpHeader() {
        parentActivity.header.clear()
    }
}